<launch>

    <arg name="show_output"   default="false" />
    <arg name="publish_rate"  default="15"/>
    <arg name="scale"         default="0.5"/>
    <arg name="num_colors"    default="2"/>

	<node name="bender_perception" pkg="bender_perception" type="bender_perception" required="true" output="screen">
        <param name="publish_rate" value="$(arg publish_rate)" />     <!-- default = 10 -->
        <param name="scale" value="$(arg scale)" />          <!-- default = 1.0 -->
        <param name="num_colors" value="$(arg num_colors)" />       <!-- default = 2, do not exceed 8 or computer will struggle -->
	</node>

     <node name="image_view" if="$(arg show_output)" pkg="image_view" type="image_view">
        <remap from="image" to="/bender_perception/image_quantized"/>
        <param name="autosize" value="true" />
    </node>

    <node name="image_scan_filter" pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen">
        <rosparam command="load" file="$(find bender_perception)/config/scan_filter_config.yaml" />
        <remap from="scan" to="bender_perception/scan_from_image" />
        <remap from="scan_filtered" to="bender_perception/scan_from_image_filtered" />
    </node>

</launch>